src/suction_grasp_data.cpp

```diff
@@ -0,0 +1,395 @@
+  grasp_posture_.points[0].time_from_start = ros::Duration(grasp_time_from_start);
+
+  // Copy values from RobotModel
+  ee_jmg_ = robot_model_->getJointModelGroup(end_effector_name);
+  arm_jmg_ = robot_model_->getJointModelGroup(ee_jmg_->getEndEffectorParentGroup().first);
+  parent_link_ = robot_model_->getLinkModel(ee_jmg_->getEndEffectorParentGroup().second);
+
+  ROS_INFO_NAMED("grasp_data", "ee_name: %s, arm_jmg: %s, parent_link: %s", ee_jmg_->getName().c_str(),
+                 arm_jmg_->getName().c_str(), parent_link_->getName().c_str());
+
+  if (define_tcp_by_name)
+  {
+    robot_state::RobotState state(robot_model_);
+    state.setToDefaultValues();
+    state.update();
+    if (!state.knowsFrameTransform(parent_link_->getName()))
+    {
+      ROS_ERROR_NAMED("grasp_data", "Robot Model does not know the frame transform for the end effector group parent "
+                                    "frame: %s. Did you set a parent link in the srdf?",
+                      parent_link_->getName().c_str());
+    }
+    if (!state.knowsFrameTransform(tcp_name_))
+    {
+      ROS_ERROR_NAMED("grasp_data", "Robot Model does not know the frame transform for the tcp frame: %s. Is it "
+                                    "available in the urdf?",
+                      tcp_name_.c_str());
+    }
+    Eigen::Isometry3d eef_mount_pose = state.getGlobalLinkTransform(parent_link_);
+    Eigen::Isometry3d tcp_mount_pose = state.getGlobalLinkTransform(tcp_name_);
+    tcp_to_eef_mount_ = tcp_mount_pose.inverse() * eef_mount_pose;
+  }
+
+  return true;
+}
+
+bool GraspData::setRobotStatePreGrasp(robot_state::RobotStatePtr& robot_state)
+{
+  ROS_WARN_STREAM_NAMED("grasp_data", "setRobotStatePreGrasp is probably wrong");
+  return setRobotState(robot_state, pre_grasp_posture_);
+}
+
+bool GraspData::setRobotStateGrasp(robot_state::RobotStatePtr& robot_state)
+{
+  ROS_WARN_STREAM_NAMED("grasp_data", "setRobotStateGrasp is probably wrong");
+  return setRobotState(robot_state, grasp_posture_);
+}
+
+bool GraspData::setRobotState(robot_state::RobotStatePtr& robot_state, const trajectory_msgs::JointTrajectory& posture)
+{
+  // Assume joint trajectory has only 1 waypoint
+  if (end_effector_type_ == FINGER)
+  {
+    if (posture.points.size() < 1)
+    {
+      ROS_ERROR_STREAM_NAMED("grasp_data", "Posture trajectory for finger'd grasper must have at least 1 waypoint");
+      return false;
+    }
+
+    // TODO(davetcoleman): make this more efficient
+    // Do for every joint in end effector
+    for (std::size_t i = 0; i < posture.joint_names.size(); ++i)
+    {
+      // Set joint position
+      robot_state->setJointPositions(posture.joint_names[i], posture.points[0].positions);
+    }
+  }
+  return true;
+}
+
+bool GraspData::setGraspWidth(const double& percent_open, const double& min_finger_width,
+                              trajectory_msgs::JointTrajectory& grasp_posture)
+{
+  if (percent_open < 0 || percent_open > 1)
```

Comment:
Since the variable is called `percent_open`, I would have tried to input values in range [0, 100]. Should the checks be for less than zero and greater than 100 instead?